h2o-algos/src/main/java/hex/gam/GAM.java

```diff
@@ -444,6 +445,8 @@ public void assertValidGAMColumnsCountSplineTypes() {
         if (_parms._bs[index] == CS_SPLINE_TYPE)
           _cubicSplineNum++;
         if (_parms._bs[index] == IS_SPLINE_TYPE) {
+          if (!IRLSM.equals(_parms._solver))
```

Comment:
If I'm reading the code correctly, this is evaluated only when called from `init(expensive=true)`. I think checking for things like this should work even for `init(expensive=false)`.

I think this is useful for flow - we can validate the parameters before we start the training and I think we use it in automl somewhere too to validate if the parameters were correctly set.

Reply:
This is no longer relevant.  I move the warning to be before expensive = true per your suggestion.